package NYT;

option java_package = "ru.yandex.kernel.yt.protos.NYT";
option java_outer_classname = "Extension";

import "google/protobuf/descriptor.proto";

// Enum is named a little funky, because we are putting it directly into NYT namespace and
// want to avoid name conflicts. Actual string 'ETableOptimizeFor' is not used in proto files anyway.
enum ETableOptimizeFor {
    OF_LOOKUP = 0;
    OF_SCAN = 1;
};

extend google.protobuf.MessageOptions {
    optional ETableOptimizeFor optimize_for = 60223 [ default = OF_LOOKUP ];
}

extend google.protobuf.FieldOptions {
    optional bool key = 60123 [ default = false ];
    optional string lock = 60124;
    optional string expression = 60125;
    optional string aggregate = 60126;
    optional string group = 60127;
    // enables serialization if embedded messages is yson instead of protobuf serialization
    // primary use is to store embedded messages in ANY type fields on YT (if used with [NYT.flags=ANY] extension)
    optional bool message_to_yson = 60128 [ default = false ];
}
